packages/react-query/src/__tests__/useQueries.test.tsx

```diff
@@ -1104,4 +1104,55 @@ describe('useQueries', () => {
       await waitFor(() => rendered.getByText('error boundary'))
     })
   })
+
+  it("should throw error if in one of queries' queryFn throws and useErrorBoundary is in use", async () => {
+    const key1 = queryKey()
+    const key2 = queryKey()
+    const key3 = queryKey()
+
+    function Page() {
+      useQueries({
+        queries: [
+          {
+            queryKey: key1,
+            queryFn: () =>
+              Promise.reject(
+                new Error(
+                  'this should not throw because useErrorBoundary is not set',
+                ),
+              ),
+          },
+          {
+            queryKey: key2,
+            queryFn: () => Promise.reject(new Error('single query error')),
+            useErrorBoundary: true,
```

Comment:
can we also have a test where `useErrorBoundary` is a function?

Reply:
[3fd6426](https://github.com/TanStack/query/pull/4177/commits/3fd6426d0ccf7c200a8e715d22aa69d98a57b886)